alerting/src/main/kotlin/org/opensearch/alerting/util/CrossClusterMonitorUtils.kt

```diff
@@ -0,0 +1,231 @@
+            return isRemoteMonitor(monitor = monitor, localClusterName = clusterService.clusterName.value())
+        }
+
+        /**
+         * Parses the list of indexes into a map of CLUSTER_NAME to List<INDEX_NAME>.
+         * @param indexes A list of index names in "<CLUSTER_NAME>:<INDEX_NAME>" format.
+         * @param localClusterName The name of the local cluster.
+         * @return A map of CLUSTER_NAME to List<INDEX_NAME>
+         */
+        @JvmStatic
+        fun separateClusterIndexes(indexes: List<String>, localClusterName: String): HashMap<String, MutableList<String>> {
+            val output = hashMapOf<String, MutableList<String>>()
+            indexes.forEach { index ->
+                var clusterName = parseClusterName(index)
+                val indexName = parseIndexName(index)
+
+                // If the index entry does not have a CLUSTER_NAME, it indicates the index is on the local cluster.
+                if (clusterName.isEmpty()) clusterName = localClusterName
+
+                output.getOrPut(clusterName) { mutableListOf() }.add(indexName)
+            }
+            return output
+        }
+
+        /**
+         * Parses the list of indexes into a map of CLUSTER_NAME to List<INDEX_NAME>.
+         * @param indexes A list of index names in "<CLUSTER_NAME>:<INDEX_NAME>" format.
+         *      Local indexes can also be in "<INDEX_NAME>" format.
+         * @param clusterService Used to retrieve the name of the local cluster.
+         * @return A map of CLUSTER_NAME to List<INDEX_NAME>
+         */
+        @JvmStatic
+        fun separateClusterIndexes(indexes: List<String>, clusterService: ClusterService): HashMap<String, MutableList<String>> {
+            return separateClusterIndexes(indexes = indexes, localClusterName = clusterService.clusterName.value())
+        }
+
+        /**
+         * The [NodeClient] used by the plugin cannot execute searches against local indexes
+         * using format "<LOCAL_CLUSTER_NAME>:<INDEX_NAME>". That format only supports querying remote indexes.
+         * This function formats a list of indexes to be supplied directly to a [SearchRequest].
+         * @param indexes A list of index names in "<CLUSTER_NAME>:<INDEX_NAME>" format.
+         * @param localClusterName The name of the local cluster.
+         * @return A list of indexes with any remote indexes in "<CLUSTER_NAME>:<INDEX_NAME>" format,
+         *      and any local indexes in "<INDEX_NAME>" format.
+         */
+        @JvmStatic
+        fun parseIndexesForRemoteSearch(indexes: List<String>, localClusterName: String): List<String> {
+            return indexes.map {
+                var index = it
+                val clusterName = parseClusterName(it)
+                if (clusterName.isNotEmpty() && clusterName == localClusterName) {
+                    index = parseIndexName(it)
+                }
+                index
+            }
+        }
```

Comment:
This method is confusing to me, in the end it seems only useful to user input `<local cluster name>:<index>` and transform it in to `<index>`
Instead of doing such thing, why not just let it fail?

Reply:
If the local cluster is named `cluster-1`, and a monitor input is configured with indexes `["cluster-1:index-1","cluster-2:index-2"]`, the node client of the local cluster will not be able to query the first index because the pattern references the local cluster name. This function will parse that list of indexes to `["index-1","cluster-2:index-2"]` so it can be successfully fed into the node client.

Reply:
Yeah, I get this function. The confusing part is why we want to support user to use local cluster name in the input, does the other feature from UI also support that?

Reply:
I wanted to cover cases where a user may do this without knowing.

Reply:
If user cannot do it in other place, this should also be forbidden here.
So to decide, try do a search call in a cross cluster environment with `<local cluster name>:<index>` on a local index, and see if that works. If so, this is ok.